sklearn/compose/_column_transformer.py

```diff
@@ -0,0 +1,591 @@
+                raise AttributeError("Transformer %s (type %s) does not "
+                                     "provide get_feature_names."
+                                     % (str(name), type(trans).__name__))
+            feature_names.extend([name + "__" + f for f in
+                                  trans.get_feature_names()])
+        return feature_names
+
+    def _update_fitted_transformers(self, transformers):
+        transformers = iter(transformers)
+        transformers_ = []
+
+        for name, old, column in self.transformers:
+            if old == 'drop':
+                trans = old
+            elif old == 'passthrough':
+                # FunctionTransformer is present in list of transformers,
+                # so get next transformer, but save original string
+                next(transformers)
+                trans = old
+            else:
+                trans = next(transformers)
+
+            transformers_.append((name, trans, column))
+
+        self.transformers_ = transformers_
+
+    def _fit_transform(self, X, y, func, fitted=False):
+        """
+        Private function to fit and/or transform on demand.
+
+        Return value (transformers and/or transformed X data) depends
+        on the passed function.
+        ``fitted=True`` ensures the fitted transformers are used.
+        """
+        try:
+            return Parallel(n_jobs=self.n_jobs)(
+                delayed(func)(clone(trans) if not fitted else trans,
+                              X_sel, y, weight)
+                for name, trans, X_sel, weight in self._iter(
+                    X=X, fitted=fitted, replace_strings=True))
+        except ValueError as e:
+            if "Expected 2D array, got 1D array instead" in str(e):
+                raise ValueError(_ERR_MSG_1DCOLUMN)
+            else:
+                raise
+
+    def fit(self, X, y=None):
+        """Fit all transformers using X.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        y : array-like, shape (n_samples, ...), optional
+            Targets for supervised learning.
+
+        Returns
+        -------
+        self : ColumnTransformer
+            This estimator
+
+        """
+        self._validate_transformers()
+        self._validate_unspecified(X)
+
+        transformers = self._fit_transform(X, y, _fit_one_transformer)
+
+        self._update_fitted_transformers(transformers)
+        return self
+
+    def fit_transform(self, X, y=None):
+        """Fit all transformers, transform the data and concatenate results.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        y : array-like, shape (n_samples, ...), optional
```

Comment:
Is this the standard for shape of y now? I haven't seen that before.

Reply:
This is how it is documented in Pipeline. However in the transformers this is often just left out or indicated as 'ignored'. Not sure what would be best here.